### DISCUSSION TITLE
Performance Dead slow in project with Large Images - Mac OS Desktop App

### DISCUSSION THREAD

**Discussion by ccgeraldes** [jgraph/drawio#3904] [Q&A]
Hello, I had to import a drawing from Visio. I've been working on it and the images are high resolution.

So the project is really heavy and Draw.io does not have capacity to handle. Performance is very lagged like the machine is almost crashing.

I have a powerful m1 machine, does anyone know how to reduce project filesize and improve performance?

**Comment by davidjgraph** [comment]
What is the drawio version? Is it possible to share the file?

**Reply by ccgeraldes** [reply]
Unfortunately I can't share the file due to privacy policy. But is unusable :( 

22.0.0

**Reply by ccgeraldes** [reply]
Unfortunatly I just upgraded to 22.0.2 and is the same

**Comment by ccgeraldes** [comment]
![image](https://github.com/jgraph/drawio/assets/146839533/e6a0d692-bdf2-42e0-a008-414067bd2c81)

Cpu spike when moving any object or copy and pasting

**Reply by ccgeraldes** [reply]
@davidjgraph any chance I send you the file but privately ? I can't share it pubicly and I really need help

**Comment by quaryum** [comment]
I have the same problem on my windows machine, there is no visual in my design, everything is drawing but the project does not move, freezes, is there a limit? I can send you the project. My machine is amd ryzen9, 32 gb ram, nvidia 1080 video card and hdd nvme.  
[Sunucu Odası Diagramı 2023.zip](https://github.com/jgraph/drawio/files/12858686/Sunucu.Odasi.Diagrami.2023.zip)

**Reply by davidjgraph** [reply]
Running what desktop build?

**Reply by quaryum** [reply]
After upgrading to Drawio v22.0.2 I no longer have performance problems, thank you very much

**Reply by ccgeraldes** [reply]
After upgrading I still have the same issue. I can't use this project.

Can someone please help? I can send the project privately to someone I really need help here
